packages/narwallets/src/lib/narwallets.ts

```diff
@@ -0,0 +1,274 @@
+  iconUrl?: string;
+  deprecated?: boolean;
+}
+
+interface PendingPromises {
+  id_wallet_selector: number;
+  code: string;
+  // Note: the result from different action may give different results, like boolean or FinalExecutionOutcome. Since the response may be many things, we're leaving any here
+  resolve: (value: any) => void;
+  reject: (reason?: any) => void;
+  timeout?: NodeJS.Timeout;
+}
+
+const NARWALLETS_CODES = {
+  SIGN_IN: "sign-in",
+  IS_INSTALLED: "is-installed",
+  IS_SIGNED_IN: "is-signed-in",
+  SIGN_OUT: "sign-out",
+  GET_ACCOUNT_ID: "get-account-id",
+  SIGN_AND_SEND_TRANSACTION: "sign-and-send-transaction",
+  SIGN_AND_SEND_TRANSACTIONS: "sign-and-send-transactions" 
+}
+
+let id = 0;
+const pendingPromises: Array<PendingPromises> = [];
+
+const sendToNarwallets = (
+  code: string,
+  withTimeout = false,
+  params?: any
+): Promise<any> => {
+  const promise = new Promise<any>((resolve, reject) => {
+    id++;
+    let promiseTimeout;
+    if (withTimeout) {
+      promiseTimeout = setTimeout(() => {
+        return reject(Error("timeout"));
+      }, 2000);
+    }
+    pendingPromises.push({
+      id_wallet_selector: id,
+      code,
+      resolve,
+      reject,
+      timeout: promiseTimeout,
+    });
+
+    window.postMessage({
+      id,
+      src: "ws",
+      type: "nw",
+      code,
+      dest: "ext",
+      params,
+    });
+  });
+  return promise;
+};
+
+const isInstalled = async (): Promise<boolean> => {
+  // Note: sendToNarwallets throws if not installed
+  try {
+    await sendToNarwallets(NARWALLETS_CODES.IS_INSTALLED, true);
+  }
+  catch (ex) {
+    return false
+  }
+  return true;
+};
+
+const isSignedIn = (): Promise<boolean> => {
+  return sendToNarwallets(NARWALLETS_CODES.IS_SIGNED_IN, true);
+};
+
+const getAccountId = (): Promise<string> => {
+  return sendToNarwallets(NARWALLETS_CODES.GET_ACCOUNT_ID, true);
+};
+
+const callSignAndSendTransaction = (
+  params: SignAndSendTransactionParams
+): Promise<providers.FinalExecutionOutcome> => {
+  return sendToNarwallets(NARWALLETS_CODES.SIGN_AND_SEND_TRANSACTION, false, params);
+};
+
+const callSignAndSendTransactions = (
+  params: Array<SignAndSendTransactionParams>
+): Promise<
+  Array<providers.FinalExecutionOutcome>
+> => {
+  return sendToNarwallets(NARWALLETS_CODES.SIGN_AND_SEND_TRANSACTIONS, false, params);
+};
+
+function findPendingPromiseById(id: number): PendingPromises | undefined {
+  return pendingPromises.filter((c) => c.id_wallet_selector == id)[0];
+}
+
+function removePendingPromise(callback: PendingPromises) {
+  const index = pendingPromises.indexOf(callback);
+  if (index > -1) {
+    // only splice array when item is found
+    pendingPromises.splice(index, 1); // 2nd parameter means remove one item only
+  }
+}
+
+const setupNarwalletsState = (): void => {
+
+  // receive response from the extension content_script
+  window.addEventListener("message", (event) => {
+    if (event.source != window) {
```

Comment:
Can you slightly change the syntax here and use the strict version of not equals:

So instead of `!=` use `!==`
```ts
if (event.source !== window) {}
```

This also applies in cases where you have double equals so:

Instead of `==` use `===`

Reply:
Done

Reply:
I recall having pushed, but I'll do it again soon. We've fixed an issue we noticed. We're testing it properly before pushing again.

Regarding the linting issues, just added the format on save.